decoder/vaapidecoder_mpeg2.cpp

```diff
@@ -517,65 +517,18 @@ YamiStatus VaapiDecoderMPEG2::fillConfigBuffer()
         return status;
     }
 
-    m_configBuffer.width
+    uint32_t width
         = (m_sequenceExtension->horizontal_size_extension & 0x3)
           | (m_sequenceHeader->horizontal_size_value & 0xFFF);
-    m_configBuffer.height = (m_sequenceExtension->vertical_size_extension & 0x3)
+    uint32_t height = (m_sequenceExtension->vertical_size_extension & 0x3)
                             | (m_sequenceHeader->vertical_size_value & 0xFFF);
 
-    if (m_VAStart) {
-        // sequence start extension code with VA started has to conduct a
-        // port reconfiguration when stream size changes
-        if (m_configBuffer.width > m_videoFormatInfo.width
-            || m_configBuffer.height > m_videoFormatInfo.height) {
-            // need to re-start VA to generate new surfaces
-            status = VaapiDecoderBase::terminateVA();
-            if (status != YAMI_SUCCESS) {
-                return status;
-            }
-            m_VAStart = false;
-        }
-        else if (m_configBuffer.width != m_videoFormatInfo.width
-                 || m_configBuffer.height != m_videoFormatInfo.height) {
-            // VA surfaces can be re-used and client is notified
-            // public member from base class
-            m_videoFormatInfo.width = m_configBuffer.width;
-            m_videoFormatInfo.height = m_configBuffer.height;
-            m_videoFormatInfo.surfaceWidth = m_configBuffer.width;
-            m_videoFormatInfo.surfaceHeight = m_configBuffer.height;
-            m_configBuffer.surfaceWidth = m_configBuffer.width;
-            m_configBuffer.surfaceHeight = m_configBuffer.height;
-            status = YAMI_DECODE_FORMAT_CHANGE;
-        }
+    if (setFormat(width, height, width, height, kMinSurfaces + 1)) {
+        m_VAStart = false;
+        return YAMI_DECODE_FORMAT_CHANGE;
```

Comment:
setFormat doesn't check for width and height being greater than m_videoFormatInfo. As such it could be that the information doesn't get updated and VA is not restarted as it should.  It looked strange that passing same width and height twice to setFormat will produce a format change.  Remember that VA doesn't need to be restarted when surfaces can be re-used.